Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift

```diff
@@ -82,42 +82,14 @@ final class CarPlayAreasViewModel {
         devicesAndAreas: [HADeviceAreaResponse],
         server: Server
     ) {
-        /// area_id : [device_id]
-        var areasAndDevicesDict: [String: [String]] = [:]
-
-        // Get all devices from an area
-        devicesAndAreas.forEach { device in
-            if let areaId = device.areaId, let deviceId = device.deviceId {
-                if var deviceIds = areasAndDevicesDict[areaId] {
-                    deviceIds.append(deviceId)
-                    areasAndDevicesDict[areaId] = deviceIds
-                } else {
-                    areasAndDevicesDict[areaId] = [deviceId]
-                }
-            }
-        }
-
-        /// area_id : [entity_id]
-        var areasAndEntitiesDict: [String: [String]] = [:]
-
-        // Get all entities from an area
-        areasAndEntities.forEach { entity in
-            if let areaId = entity.areaId, let entityId = entity.entityId {
-                if var entityIds = areasAndEntitiesDict[areaId] {
-                    entityIds.append(entityId)
-                    areasAndEntitiesDict[areaId] = entityIds
-                } else {
-                    areasAndEntitiesDict[areaId] = [entityId]
-                }
-            }
-        }
-
+        let areasAndDevicesDict = mapToareasAndEntities(devicesAndAreas: devicesAndAreas)
+        var areasAndEntitiesDict = mapToAreasAndEntitiesDict(areasAndEntities: areasAndEntities)
```

Comment:
Have you checked if the behavior is still the same? It looks good, but worth checking. If you dont see carplay right away you can manually add `com.apple.developer.carplay-driving-task` entitlement to test it out (just dont commit it)

Reply:
Sure, that would be really good

Reply:
If you prefer you can do it on a separate PR and add //swiftlint:disable cyclomatic_complexity for now in this class, so we can keep the PR topic in focus

Reply:
I've reverted the change and disable the swiftlint rule.